src/software/ai/navigator/path_planner/simulated_theta_star_test.cpp

```diff
@@ -0,0 +1,241 @@
+                              ValidationCoroutine::push_type& yield) {
+            Rectangle expectedFinalPosition(Point(0.985, -0.26), Point(1.15, -0.3));
+            robotInPolygon(1, expectedFinalPosition, world_ptr, yield);
+        }};
+
+    std::vector<ValidationFunction> non_terminating_validation_functions = {};
+
+    runTest(terminating_validation_functions, non_terminating_validation_functions,
+            Duration::fromSeconds(15));
+}
+
+TEST_F(SimulatedThetaStarTest, test_theta_star_robot_in_obstacle)
+{
+    // Destination is in a free point, but initial point is in an obstacle
+    Point destination      = Point(0, 0);
+    Point initial_position = Point(1, 0);
+    setBallState(BallState(Point(0, -0.6), Vector(0, 0)));
+    addFriendlyRobots(
+        TestUtil::createStationaryRobotStatesWithId({Point(-3, 0), initial_position}));
+    addEnemyRobots(TestUtil::createStationaryRobotStatesWithId({Point(1, 0)}));
+
+    auto tactic = std::make_shared<MoveTactic>(false);
+    tactic->updateControlParams(destination, Angle::zero(), 0);
+    setTactic(tactic);
+    setRobotId(1);
+
+    std::set<MotionConstraint> motion_constraints;
+    motion_constraints.insert(MotionConstraint::ENEMY_ROBOTS_COLLISION);
+    setMotionConstraints(motion_constraints);
+
+
+    std::vector<ValidationFunction> terminating_validation_functions = {
+        [destination, tactic](std::shared_ptr<World> world_ptr,
+                              ValidationCoroutine::push_type& yield) {
+            Rectangle expectedFinalPosition(Point(0.015, 0.015), Point(-0.015, -0.015));
+            robotInPolygon(1, expectedFinalPosition, world_ptr, yield);
+        }};
+
+    std::vector<ValidationFunction> non_terminating_validation_functions = {};
+
+    runTest(terminating_validation_functions, non_terminating_validation_functions,
+            Duration::fromSeconds(15));
+}
+
+TEST_F(SimulatedThetaStarTest, test_theta_no_obstacle_straight_path)
+{
+    Point destination      = Point(-2, 1);
+    Point initial_position = Point(2, 1);
+    setBallState(BallState(Point(0, -0.6), Vector(0, 0)));
+    addFriendlyRobots(
+        TestUtil::createStationaryRobotStatesWithId({Point(-3, 0), initial_position}));
+    addEnemyRobots(TestUtil::createStationaryRobotStatesWithId({Point(1, 0)}));
+
+    auto tactic = std::make_shared<MoveTactic>(false);
+    tactic->updateControlParams(destination, Angle::zero(), 0);
+    setTactic(tactic);
+    setRobotId(1);
+
+    std::set<MotionConstraint> motion_constraints;
+    motion_constraints.insert(MotionConstraint::ENEMY_ROBOTS_COLLISION);
+    setMotionConstraints(motion_constraints);
+
+
+    std::vector<ValidationFunction> terminating_validation_functions = {
+        [destination, tactic](std::shared_ptr<World> world_ptr,
+                              ValidationCoroutine::push_type& yield) {
+            Rectangle expectedFinalPosition(Point(2.015, 1.015), Point(1.985, 0.985));
```

Comment:
snake_case please and similarly for all the instances

Reply:
Updated!